--- numpy-1.10.1/numpy/distutils/fcompiler/gnu.py.orig	2015-10-13 19:32:27.197631590 +0200
+++ numpy-1.10.1/numpy/distutils/fcompiler/gnu.py	2015-10-13 19:48:20.714278382 +0200
@@ -59,10 +59,10 @@
                     return ('gfortran', m.group(1))
         else:
             # Output probably from --version, try harder:
-            m = re.search(r'GNU Fortran\s+95.*?([0-9-.]+)', version_string)
+            m = re.search(r'GNU Fortran\s+95.*?(\d[0-9-.]*)', version_string)
             if m:
                 return ('gfortran', m.group(1))
-            m = re.search(r'GNU Fortran.*?\-?([0-9-.]+)', version_string)
+            m = re.search(r'GNU Fortran.*?\-?(\d[0-9-.]*)', version_string)
             if m:
                 v = m.group(1)
                 if v.startswith('0') or v.startswith('2') or v.startswith('3'):
